core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -194,6 +197,64 @@ class AdminManager(val config: KafkaConfig,
     }
   }
 
+  def createPartitions(timeout: Int,
+                       newPartitions: Map[String, NewPartitions],
+                       validateOnly: Boolean,
+                       listenerName: ListenerName,
+                       callback: Map[String, ApiError] => Unit): Unit = {
+    var waiting = Map.empty[String, NewPartitions]
+    var done = Map.empty[String, ApiError]
+
+    newPartitions.foreach{case (topic, newPartition: NewPartitions) =>
+      try {
+        val oldNumPartitions = zkUtils.getTopicPartitionCount(topic) match {
+          case Some(count) => count
+          case None => throw new InvalidTopicException()
+        }
+        val newNumPartitions = newPartition.totalCount
+        val numPartitionsIncrement = newNumPartitions - oldNumPartitions
+        if (numPartitionsIncrement <= 0) {
+          throw new InvalidPartitionsException(s"Topic currently has $oldNumPartitions partitions: Cannot create $numPartitionsIncrement partitions.")
```

Comment:
Maybe this should say something like s"Topic currently has $oldNumPartitions, which is higher than requested $newNumPartitions"?